src/client/datascience/interactive-ipynb/trustCommandHandler.ts

```diff
@@ -65,10 +65,14 @@ export class TrustCommandHandler implements IExtensionSingleActivationService {
                 sendTelemetryEvent(Telemetry.TrustAllNotebooks);
                 break;
             case DataScience.trustNotebook():
-                if (model instanceof VSCodeNotebookModel) {
-                    await this.trustNativeNotebook(model);
-                } else {
-                    await this.trustNotebook(model);
+                try {
+                    if (model instanceof VSCodeNotebookModel) {
+                        await this.trustNativeNotebook(model);
+                    } else {
+                        await this.trustNotebook(model);
+                    }
+                } catch (e) {
+                    traceError('Error while trusting notebook', e);
```

Comment:
Why would this fail?

Reply:
Currently they shouldn't, but there's nothing enforcing that either trustNativeNotebook or trustNotebook cannot throw.